### PR TITLE
cplp-907 enhancing backend service for welcome email

### PR DESCRIPTION
**Pull request by Akshay-Chaudhari-BMW**
enhancing backend service for welcome email

### CODE REVIEW

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/PortalBackendDBAccess.cs [hunk 1]
```diff
@@ -341,6 +341,18 @@ public async IAsyncEnumerable<CompanyRoleData> GetCompanyRoleAgreementsUntracked
                 .AsNoTracking()
                 .AsAsyncEnumerable();
 
+        public IAsyncEnumerable<WelcomeEmailData> GetWelcomeEmailDataUntrackedAsync(Guid applicationId) =>
+        (from invitation in _dbContext.Invitations
+         join companyuser in _dbContext.CompanyUsers on invitation.CompanyUserId equals companyuser.Id
+         join company in _dbContext.Companies on companyuser.CompanyId equals company.Id
+         where invitation.CompanyApplicationId == applicationId
+         select new WelcomeEmailData(
+                     companyuser.Firstname + " " + companyuser.Lastname,
```
Comment (ntruchsess) [inline src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/PortalBackendDBAccess.cs:350]: both FirstName and LastName may be null here

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/PortalBackendDBAccess.cs [hunk 2]
```diff
@@ -341,6 +341,18 @@ public async IAsyncEnumerable<CompanyRoleData> GetCompanyRoleAgreementsUntracked
                 .AsNoTracking()
                 .AsAsyncEnumerable();
 
+        public IAsyncEnumerable<WelcomeEmailData> GetWelcomeEmailDataUntrackedAsync(Guid applicationId) =>
+        (from invitation in _dbContext.Invitations
```
Comment (ntruchsess) [inline src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/PortalBackendDBAccess.cs:345]: no need to join Invitations - you are not reading data from there and the referential integrity with invitations is taken care of by a database-constraint.

File: src/administration/CatenaX.NetworkServices.Administration.Service/BusinessLogic/UserBusinessLogic.cs [hunk 3]
```diff
@@ -238,17 +238,26 @@ public async Task<bool> AddBpnAttributeAsync(IEnumerable<UserUpdateBpn>? usersTo
         }
 
         //TODO: full functionality is not yet delivered and currently the service is working with a submitted Json file
-        public async Task<bool> PostRegistrationWelcomeEmailAsync(WelcomeData welcomeData)
+        public async Task<bool> PostRegistrationWelcomeEmailAsync(Guid applicationId)
         {
-            var mailParameters = new Dictionary<string, string>
+            var userDetails = await _portalDBAccess.GetWelcomeEmailDataUntrackedAsync(applicationId).ToListAsync().ConfigureAwait(false);
```
[the remark this answers is not in the document]
Comment (Akshay-Chaudhari-BMW) [reply src/administration/CatenaX.NetworkServices.Administration.Service/BusinessLogic/UserBusinessLogic.cs]: removed ToListAsync()

File: src/administration/CatenaX.NetworkServices.Administration.Service/BusinessLogic/UserBusinessLogic.cs [hunk 4]
```diff
@@ -238,17 +238,26 @@ public async Task<bool> AddBpnAttributeAsync(IEnumerable<UserUpdateBpn>? usersTo
...
+                {
+                    { "userName", user.UserName },
+                    { "companyName", user.CompanyName },
+                    { "url", $"{_settings.Portal.BasePortalAddress}"},
+                };
 
-            await _mailingService.SendMails(welcomeData.email, mailParameters, new List<string> { "EmailRegistrationWelcomeTemplate" }).ConfigureAwait(false);
+                if (String.IsNullOrWhiteSpace(user.EmailId))
+                {
+                    throw new ArgumentNullException("Email must not be empty");
```
[the remark this answers is not in the document]
Comment (Akshay-Chaudhari-BMW) [reply src/administration/CatenaX.NetworkServices.Administration.Service/BusinessLogic/UserBusinessLogic.cs]: resolved

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/PortalBackendDBAccess.cs (hunk 1 above)
Comment (Akshay-Chaudhari-BMW) [reply src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/PortalBackendDBAccess.cs]: changed to nullable

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/PortalBackendDBAccess.cs (hunk 2 above)
Comment (Akshay-Chaudhari-BMW) [reply src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/PortalBackendDBAccess.cs]: resolved

**Review by ntruchsess — CHANGES_REQUESTED** [review]

**Pull request merged**